core/domain/summary_services.py

```diff
@@ -163,6 +163,72 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_metadata_dicts(query_string, search_cursor):
+    """Given a query string and a search cursor, returns a list of exploration
+    dicts that satisfy the search query.
+
+    Args:
+        query_string: str. The search query for which the search is to be
+            performed.
+        search_cursor: The cursor location for next search.
+    Returns:
+        exploration_list: A list of metadata dicts of the exploration ids 
+            for the query_string.
+        new_search_cursor: New search cursor location.
+    """
+
+    exp_ids, new_search_cursor = (
+        exp_services.get_exploration_ids_matching_query(
+            query_string, cursor=search_cursor))
+
+    exploration_list = (get_exploration_metadata_dicts(exp_ids))
+
+    return exploration_list, new_search_cursor
+
+
+def get_exploration_metadata_dicts(exploration_ids, editor_user_id=None):
+    """Given a list of exploration ids, optionally filters the list for
+    explorations that are currently non-private and not deleted, and returns a
+    list of dicts of the corresponding exploration summaries for collection
+    node search.
+    
+    Args:
+        exploration_ids: A list of exploration ids for which exploration
+            metadata dicts are to be returned.
+        editor_user_id: str or None. User id of the editor to check for users
```

Comment:
str or None. If not None, private explorations that are editable by this user are also returned.